src/cli/cli.cpp

```diff
@@ -0,0 +1,123 @@
+#include "cli.hpp"
+#include "states.hpp"
+#include "Particle.h"
+#include "conio.hpp"
+#include "menuItems/systemCommands.hpp"
+
+#include <fstream>
+
+const CLI_menu_t CLI_menu[] =
+{
+    {"menu", "display Menu", &CMD_displayMenu},
+    {"disconnect", "disconnect particle", &CLI_disconnect},
+    {"connect", "connect particle", &CLI_connect},
+    {"showerrors", "show flog errors", &CLI_displayFLOG},
+    {NULL,NULL,NULL}
+};
+
+static STATES_e CLI_nextState;
+
+void CLI::init(void) 
+{
+    CLI_nextState = STATE_CLI;
+
+    // While there is an avaliable character typed, get it
+    while(kbhit())
+    {
+        getch();
+    }
+}
+
+STATES_e CLI::run(void)
+{
+    CLI_menu_t *cmd;
+    char* userInput = new char[100];
+    userInput[0] = 0;
+
+    SF_OSAL_printf("\n>");
```

Comment:
We should define `consts.hpp` and define the newline convention for this project as a compile-time macro.

Since you are doing this, I suggest `#define NEWLINE "\n"` or `#define __NL__ "\n"`, and use implicit string concatenation to include, e.g. `printf(__NL__ ">")`.

Reply:
Done